containers/themes/ThemesSection.js

```diff
@@ -0,0 +1,108 @@
+import React, { useState } from 'react';
+import { c } from 'ttag';
+import {
+    useMailSettings,
+    useEventManager,
+    useApiWithoutResult,
+    SubTitle,
+    Alert,
+    ThemeCards,
+    ConfirmModal,
+    TextArea
+} from 'react-components';
+import { updateTheme } from 'proton-shared/lib/api/mailSettings';
+import { themeDark, themeLight, themeBlue, themeCustom } from './availableThemes.js';
+
+const availableThemes = [themeDark, themeLight, themeBlue, themeCustom];
+
+const getThemeName = (Theme) => {
+    if (Theme) {
+        if (/\/* light-theme \/*/.test(Theme)) {
+            return 'light-theme';
+        }
+        if (/\/* blue-theme \/*/.test(Theme)) {
+            return 'blue-theme';
+        }
+        return 'custom-theme';
+    }
+    return 'dark-theme';
+};
+const getCSS = (themeName) => {
+    switch (themeName) {
+        case 'light-theme':
+            return CSSLightTheme;
+        case 'blue-theme':
+            return CSSBlueTheme;
+        default:
+            return '';
+    }
+};
+
+const ThemesSection = () => {
+    const [showCustomModal, setShowCustomModal] = useState(false);
+    const [customCSSInput, setCustomCSSInput] = useState('');
+
+    const [{ Theme }] = useMailSettings();
+    const { call } = useEventManager();
+    const { request, loading } = useApiWithoutResult(updateTheme);
+
+    const themeName = getThemeName(Theme);
+
+    const handleChangeTheme = async (themeName) => {
+        if (themeName === 'custom-theme') {
+            setShowCustomModal(true);
+            return;
+        }
+        setCustomCSSInput('');
+        await request(getCSS(themeName));
+        call();
+    };
+    const handleConfirm = async () => {
+        setShowCustomModal(!showCustomModal);
+        await request(customCSSInput);
+        call();
+    };
+    const handleClose = () => {
+        setShowCustomModal(false);
+    };
+    const handleChangeCSSInput = (event) => {
+        setCustomCSSInput(event.target.value);
+    };
+
+    const themeText = c('Info')
+        .t`ProtonMail offers 3 default themes to select from. You can also import a custom theme using our CSS editor`;
+    const customizationThemeText = c('Info')
+        .t`Selecting another theme will override your current theme and any customization will be lost`;
+    const customThemeText = c('Info')
+        .t`Custom themes from third parties can potentially betray your privacy. Only use themes from trusted sources`;
+
+    return (
+        <>
+            <SubTitle>{c('Title').t`Themes`}</SubTitle>
+            <Alert>{themeText}</Alert>
+            <Alert type="warning">{customizationThemeText}</Alert>
+            <br />
+            <ThemeCards list={availableThemes} themeName={themeName} onChange={handleChangeTheme} disabled={loading} />
+            <ConfirmModal
```

Comment:
Extend the `InputModal` to display an `<input />` or a `<textarea>` please.

Reply:
I did some modification to `InputModal`. In particular I added a `useEffect` so that the component would update if it's closed (otherwise it retains the input from the last time it was open, even if clicking `cancel`).

However, to match the design https://user-images.githubusercontent.com/33841139/56794123-caee7700-680d-11e9-9585-edda1b962c71.png the component should be further modified (to fit the `Alert`). I wonder if it's better to create a new component `TextAreaModal`.